exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -47,16 +47,25 @@
   private List<FileStatus> statuses;
 
   public List<String> files;
+  /**
+   * root path for the selections
+   */
   public final String selectionRoot;
+  /**
+   * root path for the metadata cache file (if any)
+   */
+  public final String cacheFileRoot;
```

Comment:
Thanks for the explanation. It makes sense and I do see why updating selectionRoot would cause problems for the queries you mentioned.